### DISCUSSION TITLE
how to add curl param --cert-type P12 in hurl

### DISCUSSION THREAD

**Discussion by Taoaozw** [Orange-OpenSource/hurl#2563] [Q&A] [answered]
i have a pfx file  and i don't know how to correctly use it

**Comment by jcamiel** [comment, marked as answer]
Hi @Taoaozw 

I'm not a super specialist of certificate but can you convert your p12 file  to a certificate and key file first:

```shell
$ openssl pkcs12 -in mycert.p12 -out file.key.pem -nocerts -nodes
$ openssl pkcs12 -in mycert.p12 -out file.crt.pem -clcerts -nokeys
```

Then use `--cert` and `--key` Hurl options :

```shell
$ hurl --cert file.crt.pem --key file.key.pem test.hurl
```

?
